webpack.config.js

```diff
@@ -1,174 +1,68 @@
-        "resolve-url-loader",
-        {
-            loader: "sass-loader",
-            options: {
-                sourceMap: true, // <-- !!IMPORTANT!! for "resolve-url-loader"
-            }
-        }
-    ]
-}
-
-
 module.exports = {
-    entry: entryPoints('./src/js'),
-
     output: {
-		filename: 'js/[name].js',
-		path: path.resolve(__dirname, './theme'),
+        path: path.resolve(__dirname, './theme'),
         assetModuleFilename: 'images/[name][ext]',
         clean: isDev ? false : {
             keep(asset) {
                 const targets = ['fonts/', 'html/', 'css/', 'images/', 'js/', '.css'];
                 return !targets.some(v => asset.includes(v));
             }
         }
-	},
+    },
+
+    resolve: {
+        alias: {
+            // aliases used in template for source assets
+            '@scripts': path.join(__dirname, 'src/js'),
+            '@styles': path.join(__dirname, 'src/scss'),
+            '@images': path.join(__dirname, 'src/images'),
+            '@fonts': path.join(__dirname, 'src/fonts'),
+        }
+    },
 
-    plugins: plugins(),
+    plugins: [
+        new HtmlBundlerPlugin({
+            // define a relative or absolute path to entry templates
+            // note: the partials dir must be outside the pages
+            entry: 'src/views/pages/',
+            outputPath: path.resolve(__dirname, './theme/html'),
+            js: {
+                // output filename of compiled JavaScript, used if `inline` option is false (defaults)
+                filename: 'js/[name].[contenthash:8].js',
+                //inline: true, // inlines JS into HTML
+            },
+            css: {
+                // output filename of extracted CSS, used if `inline` option is false (defaults)
+                //filename: 'css/[name].[contenthash:8].css',
+                filename: ({chunk}) => chunk.name === 'style' ? '[name].[contenthash:8].css' : 'css/[name].[contenthash:8].css'
+                //inline: true, // inlines CSS into HTML
+            },
+            preprocessor: 'eta', // use the template engine
+            preprocessorOptions: {
+                views: 'src/views', // path for includes in template
+            },
+            minify: 'auto', // minify html in production mode only
+        }),
+    ],
 
     externals: {
 		jquery: 'jQuery',
 	},
 
     module: {
         rules: [
-            {
-                test: /\.html$/i,
-                use: {
-                    loader: 'html-loader',
-                    options: {
-                        minimize: false,
-                        preprocessor: processNestedHtml
-                    }
-                } 
-            },
             {
                 test: /\.s[ac]ss|\.css/i,
-                use: styleLoaders()
+                use: [
+                    "css-loader",
+                    isDev ? null : "postcss-loader",
+                    "sass-loader", // minimizes generated CSS in production mode
```

Comment:
you don't need to specify `sourceMap: true` options because the HTML Bundler Plugin resolves URLs in CSS/SCSS without the `resolve-url-loader`